src/js/components/Share/ShareButtonDesktopTablet.jsx

```diff
@@ -52,6 +54,42 @@ class ShareButtonDesktopTablet extends Component {
     const googleCivicElectionId = 0;
     let kindOfShare = 'BALLOT';
     let whatAndHowMuchToShare;
+
+    // Add debug logging
+    console.log('Share button clicked');
+    console.log('Current pathname:', window.location.pathname);
+    console.log('VoterWeVoteId:', VoterStore.getVoterWeVoteId());
+
+    // Add dataLayer tracking
+    const { location: { pathname: currentPathname } } = window;
+    const page = lookupPageNameAndPageTypeDict(currentPathname);
+    
+    const dataLayerObject = {
+      event: 'shareClick',
+      userDetails: {
+        voterWeVoteId: VoterStore.getVoterWeVoteId(),
+      },
+      shareDetails: {
+        kindOfShare,
```

Comment:
I think it would be good to also capture whatAndHowMuchToShare in this shareDetails dictionary.